src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -86,6 +87,16 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         this.disposables.push(this._onNotebookControllerSelected);
         this.isLocalLaunch = isLocalLaunch(this.configuration);
     }
+    public async getInteractiveController(): Promise<VSCodeNotebookController | undefined> {
+        // Fetch the active interpreter and use the matching controller
+        const api = await this.pythonApi.getApi();
+        const activeInterpreter = await api.getActiveInterpreter();
```

Comment:
This change also makes it easier for https://github.com/microsoft/vscode-jupyter/issues/5861
Else we have to change a few places, when ideally we should have the controllers created & return from a single place.